<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Models for predicting clinical and molecular features — all.models • PrCaMethy</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Models for predicting clinical and molecular features — all.models"><meta name="description" content="Random forest and glmnet models for predicting various clinical and molecular features of patients diagnosed with prostate cancer.
The predictors are gene-level methylation estimated by gene.methylation."><meta property="og:description" content="Random forest and glmnet models for predicting various clinical and molecular features of patients diagnosed with prostate cancer.
The predictors are gene-level methylation estimated by gene.methylation."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PrCaMethy</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Introduction.html">Introduction</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Models for predicting clinical and molecular features</h1>

      <div class="d-none name"><code>all.models.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><a href="https://cran.r-project.org/web/packages/randomForest/index.html" class="external-link">Random forest</a> and <a href="https://cran.r-project.org/web/packages/glmnet/index.html" class="external-link">glmnet</a> models for predicting various clinical and molecular features of patients diagnosed with prostate cancer.
The predictors are gene-level methylation estimated by <a href="gene.methylation.html">gene.methylation</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">all.models</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>An object of class <code>list</code> of length 14.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Models are available for predicting the following features:</p><ul><li><p><code>age.continuous</code>: patient age in years</p></li>
<li><p><code>ISUP.grade</code>: International Society of Urological Pathology (ISUP) grade risk group (1-5). See <a href="https://www.prostate.org.au/testing-and-diagnosis/grading-genetics/your-gleason-score-isup-grade/#:~:text=The%20ISUP%20Grade&amp;text=It%20is%20done%20using%20prostate,the%20corresponding%20ISUP%20Grade%20Group." class="external-link">here</a> for further details.</p></li>
<li><p><code>t.category</code>: TNM tumour category (1-4), measures the size and extent of the primary tumour</p></li>
<li><p><code>psa.categorical</code>: Prostate-specific antigen (PSA) category: &lt;= 10, [10, 20), and &gt;= 20 ng/mL.</p></li>
<li><p><code>pga</code>: percentage of the genome altered was defined as PGA = (base-pair length of all genome regions with gain or loss) / 3.2 billion bases <code>*</code> 100</p></li>
<li><p><code>&lt;gene&gt;.cna.&lt;loss/gain&gt;</code>: features with <code>.cna.</code> in their name give the gene name and then identify whether there is a copy number loss or gain event.  See the Examples section for the full list of cna features.</p></li>
<li><p><code>log2p1.snvs.per.mbps</code>: single nucleotide variants (SNVs) per mega-base pairs (Mbps) with a log2(x + 1) transformation.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">all.models</span><span class="op">)</span>;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Models for predicting the following features:</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">all.models</span><span class="op">)</span>;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "age.continuous"       "ISUP.grade"           "t.category"          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "psa.categorical"      "pga"                  "CHD1.cna.loss"       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "NKX3-1.cna.loss"      "MYC.cna.gain"         "PTEN.cna.loss"       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "CDKN1B.cna.loss"      "RB1.cna.loss"         "CDH1.cna.loss"       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "TP53.cna.loss"        "log2p1.snvs.per.mbps"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Model class per feature, e.g. randomForest or glmnet:</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">all.models</span>, <span class="va">class</span><span class="op">)</span>;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $age.continuous</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "elnet"  "glmnet"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $ISUP.grade</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "multnet" "glmnet" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $t.category</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "randomForest"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $psa.categorical</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "randomForest"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $pga</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "randomForest"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $CHD1.cna.loss</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "lognet" "glmnet"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`NKX3-1.cna.loss`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "randomForest"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $MYC.cna.gain</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "randomForest"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $PTEN.cna.loss</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "lognet" "glmnet"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $CDKN1B.cna.loss</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "lognet" "glmnet"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $RB1.cna.loss</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "randomForest"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $CDH1.cna.loss</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "lognet" "glmnet"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $TP53.cna.loss</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "lognet" "glmnet"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $log2p1.snvs.per.mbps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "randomForest"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Required genes for predicting each feature:</span></span></span>
<span class="r-in"><span><span class="co"># lapply(all.models, function(x) x$xNames)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jaron Arbet, Paul C. Boutros.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

